### ISSUE TITLE
How to easily switch to a different keyboard?

### ISSUE THREAD

**Issue by locrianz** [Julow/Unexpected-Keyboard#507]
Many keyboards have a means of easily bringing up the system's keyboard switcher, but I can't find how to do it with this one. Am I missing it, or is it just not a thing?

**Comment by Spike-from-NH** [comment]
Look for the keyboard logo. (In the QWERTY layout, it is a northeasterly swipe of the Fn key.)

**Comment by Spike-from-NH** [comment]
No, I recently re-added a gesture to change layouts in Unexpected. (See #491, posts of 19-Nov.) That brings up a keyboard icon with three dots underneath. For me, the gesture at Fn does bring up the Android Settings "Select keyboard" menu.

I have had devices/configurations with no easy way to switch keyboards; I never pinned down a rule for when this happened. In some Androids, a Nav Bar icon to switch keyboards is a user setting. Glad your problem is solved for now!

**Comment by Julow** [comment]
There should be a "keyboard" icon on the top-right corner of the Fn key that pops up the keyboard switcher. If the corresponding option "Switch to the last used keyboard" (not set by default) is set, a long press will bring the keyboard switcher.

The system can instruct the keyboard to not show this button, so this might also be a system bug.

Note that you have to swipe on the Fn key, not tap it, otherwise you'd just activate the Fn modifier. The "keyboard" icons on the space bar do not bring the keyboard switcher.

**Comment by Julow** [comment]
Is this still an issue ?

**Comment by locrianz** [comment]
Sorry, yes, I lost track of this thread.

This is what the keyboard looks like for me. I'm using version 1.25.0, installed via F-Droid. There is no such icon on the Fn key, and swiping to the upper right just activates Fn.

![Uploading Screenshot_20240115-130813.png…]()

**Comment by locrianz** [comment]
Sorry, yes, I lost track of this thread.

This is what the keyboard looks like for me. I'm using version 1.25.0, installed via F-Droid. There is no such icon on the Fn key, and swiping to the upper right just activates Fn.

![Screenshot_20240115-130813](https://github.com/Julow/Unexpected-Keyboard/assets/152659636/c4c31695-378e-4c9e-bd70-48e47a0be7d7)

**Comment by locrianz** [comment]
Android 13, and Nokia (XR21 if that helps).

**Comment by Julow** [comment]
That's perhaps a stupid question but do you have any other keyboard enabled in the settings ? (accessible quickly from unexpected keyboard's launcher activity)

**Comment by Julow** [comment]
This might be a bug in the system (or a intended but unwanted behavior).
I decided to not let the system decide whether to show the switching key, 0a86f9a. The key is always present but can be removed in the "Add keys to the keyboard" option, 95bd931.

You can find a debug build here: https://github.com/Julow/Unexpected-Keyboard/actions/runs/7534606315

**Comment by Spike-from-NH** [comment]
This couldn't be installed over the previous debug build, so I had to start over with Settings, but it was worth it to try out your recent work (named custom settings and slider for vibration). Am eager to install a 1.26 on all my toys!

I'm seeing that the pixels-between-keycaps settings give different effects in Portrait vs Landscape. (I set each to 2px but get 4?px in one direction in one orientation.) Do you see it, @Julow, or shall I open an issue and give screenshots?

**Comment by maddy0314** [comment]
I have a flip phone running Android 11 Go that is having this issue. I don't think it can build the debug version without a finished .apk published (please correct me if wrong, am new to this) but would love to see this patched in an upcoming release.

**Comment by Julow** [comment]
You can find a debug build that contains the fix here: https://github.com/Julow/Unexpected-Keyboard/actions/runs/7616624915
